app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -327,13 +315,29 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         }
     }
 
+    private fun setEnginesDisabledWhileSyncing(isSyncing: Boolean) {
+        val bookmarksNameKey = getPreferenceKey(R.string.pref_key_sync_bookmarks)
+        findPreference<CheckBoxPreference>(bookmarksNameKey)?.apply {
+            isEnabled = !isSyncing
+        }
+        val historyNameKey = getPreferenceKey(R.string.pref_key_sync_history)
+        findPreference<CheckBoxPreference>(historyNameKey)?.apply {
+            isEnabled = !isSyncing
+        }
+        val loginsNameKey = getPreferenceKey(R.string.pref_key_sync_logins)
+        findPreference<CheckBoxPreference>(loginsNameKey)?.apply {
+            isEnabled = !isSyncing
```

Comment:
Optional, but this could also be written as:
```kotlin
    private fun setEnginesDisabledWhileSyncing(isSyncing: Boolean) {
        listOf(
            R.string.pref_key_sync_bookmarks,
            R.string.pref_key_sync_history,
            R.string.pref_key_sync_logins
        )
            .map { getPreferenceKey(it) }
            .map { findPreference<CheckBoxPreference>(it) }
            .forEach { it?.isEnabled = !isSyncing }
    }
```

or if you prefer a more imperative structure
```kotlin
    private fun setEnginesDisabledWhileSyncing(isSyncing: Boolean) {
        fun setEnabled(@StringRes strId: Int) {
            val key = getPreferenceKey(strId)
            findPreference<CheckBoxPreference>(key)?.apply {
                isEnabled = !isSyncing
            }
        }
        
        setEnabled(R.string.pref_key_sync_bookmarks)
        setEnabled(R.string.pref_key_sync_history)
        setEnabled(R.string.pref_key_sync_logins)
    }
```

Both communicate to the reader that the same changes are being made to each pref, and reduce the space for typos to hide.  But it's up to you if you want to change it, the original code is fine too.